src/main/java/io/cdap/plugin/gcp/bigquery/source/BigQuerySource.java

```diff
@@ -237,6 +240,12 @@ private void validateConfiguredSchema(Schema configuredSchema, FailureCollector
     com.google.cloud.bigquery.Schema bqSchema = getBQSchema(collector);
 
     FieldList fields = bqSchema.getFields();
+
+    if (configuredSchema == null) {
```

Comment:
This shouldn't result in an error. We do want to allow users to leave out a configured schema. In that case, the source should use whatever schema the table has.

For example, when the table is a macro, the schema may change from run to run as the table changes. In that case, the schema cannot be provided by the user.

What should happen is the plugin should error out in prepareRun if the bqSchema doesn't have any fields.

Reply:
That is what this does. This method will not be called from configurePipeline if configuredSchema is null. See lines 107-112; in configurePipeline, we have a null check and return early if configuredSchema == null.

Reply:
But it will be called from prepareRun and get to this failure case if the schema does not exist by the time the pipeline is run.

Reply:
no that's not what this does. This checks if the user has not given a schema and then errors. We only want to error if the BQ table doesn't have a schema.

Reply:
Basically, please check that you are able to deploy and run a pipeline where the schema is empty.